### PR TITLE
Add arm template for IoT resources

### LINKED ISSUE #91: Add clear references on the samples to setup Azure infrastructure
Samples require some Azure Setup to be executed beforehand. There are link on the samples but they're not clear enough.

### PR DESCRIPTION
**Pull request by danewalton**
- This adds an ARM template for deploying the basic IoT resources needed for our samples.

closes #91

### CODE REVIEW

File: docs/azure-bicep-deployment.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Azure IoT Bicep Deployment
+
+This guide will detail how to easily deploy the necessary Azure resources to run our samples using the command line and the Azure CLI. The following resources and skew levels will be deployed:
+
+- Azure IoT Hub (S1)
+- Azure Device Provisioning Service (S1)
```
Comment (ericwol-msft) [inline docs/azure-bicep-deployment.md:6]: azure functions?

File: docs/azure-bicep-deployment.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+az account list
+az account set --subscription <subscription id>
+
+# To see a list of locations, run the following and look for `name` values.
+az account list-locations | ConvertFrom-Json | format-table -Property name
+
+# Create a resource group
+az group create --name '<name>' --location '<location>'
+
+# Deploy the resources. This may take a couple minutes.
```
Comment (ericwol-msft) [inline docs/azure-bicep-deployment.md:45]:
```suggestion
# Deploy the resources. This will take a couple minutes.
```

**Review by ericwol-msft — APPROVED** [review]

File: docs/azure-bicep-deployment.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# Azure IoT Bicep Deployment
+
+This guide will detail how to easily deploy the necessary Azure resources to run our samples using the command line and the Azure CLI. The following resources and skew levels will be deployed:
```
Comment (vaavva) [inline docs/azure-bicep-deployment.md:3]:
```suggestion
This guide will detail how to easily deploy the necessary Azure resources to run our samples using the command line and the Azure CLI. The following resources and sku levels will be deployed:
```

File: docs/azure-bicep-deployment.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+- Azure IoT Hub (S1)
+- Azure Device Provisioning Service (S1)
+
+## Prerequisites
+
+- An Azure Account ([link here if you do not already have one](https://azure.microsoft.com/free/search/).
```
Comment (vaavva) [inline docs/azure-bicep-deployment.md:10]:
```suggestion
- An Azure Account ([link here if you do not already have one](https://azure.microsoft.com/free/search/)).
```

File: docs/azure-bicep-deployment.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+- One of the following:
+    - Have open an Azure Portal "Cloud Shell". This doesn't require any installation of further tools and may be easier for most people.
+    - The latest version of Azure CLI and Azure IoT Module
```
Comment (vaavva) [inline docs/azure-bicep-deployment.md:14]:
```suggestion
    - The latest version of Azure CLI and Azure IoT Module.
```
these end up on the same line, so either add `.` or change the formatting

**Review by vaavva — APPROVED** [review]

File: docs/azure-bicep-deployment.md (hunk 3 above)
Comment (danewalton) [reply docs/azure-bicep-deployment.md:3]: lol thank you

**Pull request merged**